File: Survival Analysis.R
library(survminer)
library(survival)

survival_unemployment1 <- read.csv(file.choose())

attach(survival_unemployment1)
str(survival_unemployment1)

time <- spell
event <- event
group <- ui

# Descriptive statistics
summary(time)
summary(event)
summary(group)
table(group)

# Kaplan-Meier non-parametric analysis
kmsurvival <- survfit(Surv(time,event) ~ 1)

summary(kmsurvival)

plot(kmsurvival, xlab="Time", ylab="Survival Probability")

ggsurvplot(kmsurvival, data=survival_unemployment1, risk.table = TRUE)


# Kaplan-Meier non-parametric analysis by group
kmsurvival1 <- survfit(Surv(time, event) ~ group)
summary(kmsurvival1)

plot(kmsurvival1, xlab="Time", ylab="Survival Probability")
ggsurvplot(kmsurvival1, data=survival_unemployment1, risk.table = TRUE)

File: SurvivalAnalysis.py
import pandas as pd
survival_unemp = pd.read_csv("C:/My Files/Excelr/08 - Survival Analysis/PythonCodes/survival_unemployment1.csv")
survival_unemp.head()
survival_unemp.describe()

survival_unemp["spell"].describe()
T = survival_unemp.spell

from lifelines import KaplanMeierFitter
kmf = KaplanMeierFitter()
kmf.fit(T,event_observed=survival_unemp.event)
kmf.plot()

survival_unemp.ui.value_counts()
kmf.fit(T[survival_unemp.ui==1], survival_unemp.event[survival_unemp.ui==1], label='1')
ax = kmf.plot()
kmf.fit(T[survival_unemp.ui==0], survival_unemp.event[survival_unemp.ui==0], label='0')
kmf.plot(ax=ax)
